articles/quickstart/backend/java-spring-security5/01-authorization.md

```diff
@@ -0,0 +1,309 @@
+
+```java
+// src/main/java/com/auth0/example/model/Message.java
+
+public class Message {
+    private final String message;
+
+    Message(String message) {
+        this.message = message;
+    }
+
+    public String getMessage() {
+        return this.message;
+    }
+}
+```
+
+Create a new class named `APIController` to handle requests to the endpoints:
+
+```java
+// src/main/java/com/auth0/example/web/APIController.java
+
+@RestController
+@RequestMapping(path = "api", produces = MediaType.APPLICATION_JSON_VALUE)
+public class APIController {
+
+    @GetMapping(value = "/public")
+    public Message publicEndpoint() {
+        return new Message("All good. You DO NOT need to be authenticated to call /api/public.");
+    }
+
+    @GetMapping(value = "/private")
+    public Message privateEndpoint() {
+        return new Message("All good. You can see this because you are Authenticated.");
+    }
+
+    @GetMapping(value = "/private-scoped")
+    public Message privateScopedEndpoint() {
+        return new Message("All good. You can see this because you are Authenticated with a Token granted the 'read:messages' scope");
+    }
+}
+```
+
+## Run and Test Your API
+
+To build and run the sample project, run the `bootRun` Gradle task.
+
+Linux or macOS:
+```bash
+./gradlew bootRun
+```
+
+Windows:
+```bash
+gradlew.cmd bootRun
+```
+
+Using a REST client such as Postman or cURL, issue a `GET` request to `http://localhost:3010/api/public`. You should receive the response:
+
+```json
+{"message":"All good. You DO NOT need to be authenticated to call /api/public."}
+```
+
+Next, issue a `GET` request to `http://localhost:3010/api/private`. You should receive a `401 Unauthorized` response.
+
+To test that your API is properly secured, you can obtain a test token in the Auth0 Dashboard:
+
+1. Go to the **Machine to Machine Applications** tab for the API you created above.
+2. Ensure that your API test application is marked as authorized.
+3. Click the **Test** tab, then **COPY TOKEN**.
+
+Issue a `GET` request to the `/api/private` endpoint, this time passing the token you obtained above as an `Authorization` header set to `Bearer YOUR-API-TOKEN-HERE`. You should then see the response:
+
+```json
+{"message":"All good. You can see this because you are Authenticated."}
+```
+
+Finally, to test that the `/api/private-scoped` is properly protected by the `read:messages` scope, make a `GET` request to the `/api/private-scoped` endpoint using the same token as above. You should see a `403 Forbidden` response, as this token does not possess the `read:messages` scope.
+
+Back in the Auth0 Dashboard:
+
+1. Go to the **Permissions** tab for the API you created above.
+2. Add a permission of `read:messages` and provide a description.
+3. Go to the **Machine to Machine Applications** tab.
+4. Expand your authorized test application, select the `read:messages` scope, then click **UPDATE** and then **CONTINUE**.
+5. Click the **Test** tab, then **COPY TOKEN**.
```

Comment:
A screenshot that encompasses these steps would be good here too.

Reply:
That's good feedback. I will look into adding screenshots to help clarify.

Reply:
Updated with screenshots.